networkx/algorithms/clique.py

```diff
@@ -18,19 +18,21 @@
 #    All rights reserved.
 #    BSD license.
 from collections import deque
-from itertools import chain, islice
+from itertools import chain
+from itertools import combinations
+from itertools import islice
 try:
     from itertools import ifilter as filter
 except ImportError:
     pass
 import networkx
-from networkx.utils.decorators import *
+from ..utils import not_implemented_for
```

Comment:
Relative imports make it clear that these imports are internal to the project, as opposed to the imports from built-in or third-party libraries. The rest of the codebase does use absolute imports, but it also is full of PEP8 violations, for example, so I don't buy that argument. As an appeal to authority, see [how Flask does it][1].

The `utils` package, containing helper functions, should not be exposed as part of the public API. See #1533. If you buy my argument there, then `from networkx.utils import ...` should not be used.

[1]: https://github.com/mitsuhiko/flask/blob/master/flask/app.py